src/metabase/query_processor/pivot.clj

```diff
@@ -211,6 +214,22 @@
       (qp/process-query-and-save-with-max-results-constraints! first-query info context)
       (qp/process-query (dissoc first-query :info) context))))
 
+(defn pivot-options
+  "Given a pivot table query and a card ID, looks at the `pivot_table.column_split` key in the card's visualization
+  settings and generates pivot-rows and pivot-cols to use for generating subqueries."
+  [query card-id]
+  (when card-id
+    (let [card              (t2/select-one :model/Card :id card-id)
```

Comment:
I'd like this better if the API endpoints in question passed in the pivot options and the `qp.pivot` namespace didn't have to worry about where they came from, there's a reason Toucan is discouraged in `metabase.query-processor.*` namespaces, and that is we're trying to keep the app DB code separate from QP code as much as possible.

It makes it much easier to test the pivot code if you can pass in the options without having to create a temp Card to do so